File: src/main/java/xyz/capybara/clamav/commands/Shutdown.java
package xyz.capybara.clamav.commands;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class Shutdown extends Command<Void> {

    public static final String COMMAND = "SHUTDOWN";

    @Override
    public String getCommandString() {
        return COMMAND;
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NULL_CHAR;
    }

    @Override
    protected Void parseResponse(String responseString) {
        log.info("Shutting down the ClamAV server");
        // no response
        return null;
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/Shutdown.kt
package xyz.capybara.clamav.commands

internal object Shutdown : Command<Unit>() {
    override val commandString
        get() = "SHUTDOWN"

    override val format
        get() = Command.CommandFormat.NULL_CHAR

    override fun parseResponse(responseString: String) = logger.info { "Shutting down the ClamAV server" }
}
